src/structural_pipeline/COMMIT-Filter/README.md

```diff
@@ -0,0 +1,82 @@
+# Documentation COMMIT-Filter
+
+The COMMIT-Filter add-on adds a post-processing pipeline step to the structural pipeline, which filters streamlines in the fiber cloud using [COMMIT2](https://github.com/daducci/COMMIT) and reconstructs connectivity matrices from the resulting filtered fiber cloud. The current integration (in `COMMIT_script.py`) is based on the [COMMIT2 example](https://github.com/daducci/COMMIT/wiki/COMMIT2) provided on the COMMIT wiki.
+
+## Installation
+Before using this add-on, ensure that COMMIT and its dependencies are installed. You can install COMMIT by following the instructions on the  [COMMIT wiki](https://github.com/daducci/COMMIT/wiki/Installation).
+
+
+## Usage
+
+To use the COMMIT-Filter add-on, follow these steps:
+
+
+1. Add the following items to your configuration file `CATO.conf` to filter the fibercloud constructed using constrained spherical deconvolution (`csd`) using information about fiber bundles form the Desikan-Killiany (`aparc`) parcellation.
+
+```
+"commit_filter":{
+  "reconstructionMethods":[
+     "csd"
+  ], 
+  "templates":[
+     "aparc"
+  ], 
+  "outputCommitDir":"OUTPUTDIR/commit", 
+  "filteredFiberFile":"OUTPUTCOMMITDIR/SUBJECT_fibers_commit_METHOD.trk", 
+  "filteredFiberPropertiesFile":"OUTPUTCOMMITDIR/SUBJECT_fiber_properties_commit_METHOD_TEMPLATE.mat", 
+  "schemeFile":"OUTPUTCOMMITDIR/dwi.scheme", 
+  "intermediateConnectomeFile":"OUTPUTCOMMITDIR/connectome.csv", 
+  "pythonInterpreter":"/usr/bin/python", 
+  "commitScriptFile":"TOOLBOXDIR/structural_pipeline/COMMIT-filter/COMMIT_script.py", 
+  "fiberWeightsFile":"OUTPUTCOMMITDIR/Results_StickZeppelinBall_COMMIT2/streamline_weights.txt", 
+  "filteredConnectivityMatrixFile":"OUTPUTCOMMITDIR/SUBJECT_connectivity_commit_METHOD_TEMPLATE.mat", 
+  "lambda":5E-4
+}
+```
+
+**Note:** If you installed COMMIT and its dependencies in a specific Andaconda environment, update the `pythonInterpreter` parameter. For example, if you are using Anaconda and the environment is called COMMIT use:
+
+
+```
+"pythonInterpreter":"/Applications/anaconda3/envs/COMMIT/bin/python", 
```

Comment:
Mac-specific, perhaps we can refer to `which python3` or for windows-users `where python3`?

Reply:
Thank you for pointing out that specifying a the python interpreter was a bit complicated. I have replaced the `pythonInterpreter` parameter with a `setupPythonScript` parameter in commit 500aa73. Now, users can set this parameter to specify the environment or leave it empty to use the default python installation. For example:

```
"setupPythonScript":"source /Applications/anaconda3/bin/activate COMMIT;"
```